### ISSUE TITLE
Incorrect interpretation of PSR-18 `ClientExceptionInterface`

### ISSUE THREAD

**Issue by acelot** [alextartan/guzzle-psr18-adapter#1]
The [PSR-18](https://www.php-fig.org/psr/psr-18/#error-handling) standard says that:

> A Client MUST NOT treat a well-formed HTTP request or HTTP response as an error condition. For example, response status codes in the 400 and 500 range MUST NOT cause an exception and MUST be returned to the Calling Library as normal.

> A Client MUST throw an instance of Psr\Http\Client\ClientExceptionInterface if and only if it is unable to send the HTTP request at all or if the HTTP response could not be parsed into a PSR-7 response object.

But this adapter transforms Guzzle `ClientException` (which thrown for [400-500 status codes too](https://guzzle.readthedocs.io/en/stable/quickstart.html#exceptions)) into PSR-18 `ClientExceptionInterface`.
```php
try {
    return $this->send($request);
} catch (GuzzleClientException $gCliEx) {
    throw ClientException::fromRequest($request, $gCliEx->getMessage(), $gCliEx->getCode());
} catch (GuzzleConnectException $gConEx) {
    throw NetworkException::fromRequest($request, $gConEx->getMessage(), $gConEx->getCode());
} catch (GuzzleRequestException $gReqEx) {
    throw RequestException::fromRequest($request, $gReqEx->getMessage(), $gReqEx->getCode());
}
```

**Comment by alextartan** [comment]
@acelot thank for pointing that up. Dunno why i missed that

I'll try to fix this today.

If you already have a solution for this, please make a PR

**Comment by alextartan** [comment]
fixed in 80666ef8fd511b69d28b182e16c59b3b9683bbe3
